manim/mobject/boolean_ops.py

```diff
@@ -0,0 +1,291 @@
+        return points
+
+    def _convert_vmobject_to_skia_path(self, vmobject: VMobject) -> SkiaPath:
+        """Converts a :class:`~.VMobject` to SkiaPath. This method only works for
+        cairo renderer because it treats the points as Cubic beizer curves.
+
+        Parameters
+        ==========
+        vmobject:
+            The :class:`~.VMobject` to convert from.
+
+        Returns
+        =======
+        SkiaPath:
+            The converted path.
+        """
+        path = SkiaPath()
+
+        if not np.all(np.isfinite(vmobject.points)):
+            points = np.zeros((1, 3))  # point invalid?
+        else:
+            points = vmobject.points
+
+        if len(points) == 0:  # what? No points so return empty path
+            return path
+
+        subpaths = vmobject.gen_subpaths_from_points_2d(points)
+        for subpath in subpaths:
+            quads = vmobject.gen_cubic_bezier_tuples_from_points(subpath)
+            start = subpath[0]
+            path.moveTo(*start[:2])
+            for p0, p1, p2, p3 in quads:
+                path.cubicTo(*p1[:2], *p2[:2], *p3[:2])
+
+            if vmobject.consider_points_equals_2d(subpath[0], subpath[-1]):
+                path.close()
+
+        return path
+
+    def _convert_skia_path_to_vmobject(self, path: SkiaPath) -> VMobject:
+        """Converts SkiaPath back to VMobject.
+        Parameters
+        ==========
+        path:
+            The SkiaPath to convert.
+
+        Returns
+        =======
+        VMobject:
+            The converted VMobject.
+        """
+        vmobject = self
+        current_path_start = np.array([0, 0, 0])
+
+        for path_verb, points in path:
+            if path_verb == PathVerb.MOVE:
+                parts = self._convert_2d_to_3d_array(points)
+                for part in parts:
+                    current_path_start = part
+                    vmobject.start_new_path(part)
+                    # vmobject.move_to(*part)
+            elif path_verb == PathVerb.CUBIC:
+                n1, n2, n3 = self._convert_2d_to_3d_array(points)
+                vmobject.add_cubic_bezier_curve_to(n1, n2, n3)
+            elif path_verb == PathVerb.LINE:
+                parts = self._convert_2d_to_3d_array(points)
+                vmobject.add_line_to(parts[0])
+            elif path_verb == PathVerb.CLOSE:
+                if config.renderer == "opengl":
+                    vmobject.close_path()
+                else:
+                    vmobject.add_line_to(current_path_start)
+            elif path_verb == PathVerb.QUAD:
+                n1, n2 = self._convert_2d_to_3d_array(points)
+                vmobject.add_quadratic_bezier_curve_to(n1, n2)
+            else:
+                raise Exception("Unsupported: %s" % path_verb)
+        return vmobject
+
+
+class Union(_BooleanOps):
+    """Union of two or more :class:`~.VMobject`s. This returns the common region of
+    the :class:`~VMobject`s.
```

Comment:
Prevent these:
![image](https://user-images.githubusercontent.com/83535735/134204075-b9fc8d06-896e-48c9-9bdf-ac66b88c19fb.png)

```suggestion
    """Union of two or more :class:`~.VMobject` s. This returns the common region of
    the :class:`~VMobject` s.
```